dubbo-configcenter/dubbo-configcenter-zookeeper/src/main/java/org/apache/dubbo/configcenter/support/zookeeper/ZookeeperDynamicConfiguration.java

```diff
@@ -107,10 +94,12 @@ protected boolean doRemoveConfig(String pathKey) throws Exception {
     @Override
     protected void doAddListener(String pathKey, ConfigurationListener listener) {
         cacheListener.addListener(pathKey, listener);
+        zkClient.addDataListener(pathKey, cacheListener, executor);
     }
 
     @Override
     protected void doRemoveListener(String pathKey, ConfigurationListener listener) {
         cacheListener.removeListener(pathKey, listener);
+        zkClient.removeDataListener(pathKey, cacheListener);
```

Comment:
What if cacheListener holds more than one DataListener? Should we check dataListener size before stop listening from zookeeper

Reply:
right.